lib/flipper/adapters/strict.rb

```diff
@@ -0,0 +1,47 @@
+module Flipper
+  module Adapters
+    # An adapter that ensures a feature exists before checking it.
+    class Strict
+      extend Forwardable
+      include ::Flipper::Adapter
+      attr_reader :name, :adapter
+
+      class NotFound < ::Flipper::Error
+        def initialize(name)
+          super "Could not find feature #{name.inspect}. Call `Flipper.add(#{name.inspect})` to create it."
+        end
+      end
+
+      HANDLERS = {
+        raise: ->(feature) { raise NotFound.new(feature.key) },
+        warn: ->(feature) { warn NotFound.new(feature.key).message },
+        noop: ->(_) { },
+      }
+
+      def_delegators :@adapter, :features, :get_all, :add, :remove, :clear, :enable, :disable
+
+      def initialize(adapter, handler: :raise)
```

Comment:
I don't love `handler`, but can't come up with anything better. `on_missing`? `behavior`? Drop the named arg entirely and just have it be the 2nd arg?

Reply:
I like `on_missing` because that's what the Strict adapter is entirely about. Then it defaults to raise, or if we include it by default, it can default to warn?

Reply:
I decided to just remove the keyword argument and use the second param as the handler name, or a block passed to the initialize method. My reasoning is that I think it's unlikely that this adapter will take other params, and we can always add this as keyword params if needed.